integrationtesting/opendcs-tests/src/test/java/org/opendcs/dao/CwmsLocationLevelDAOTestIT.java

```diff
@@ -0,0 +1,361 @@
+            
+            // Skip unit conversion test if no data exists
+            assumeTrue(baseValue != null, 
+                "Skipping unit conversion test - no data found for " + testLocationLevelId);
+            
+            // Test conversion from feet to meters
+            LocationLevelValue valueInFeet = dao.getLatestLocationLevelValue(
+                tx, testLocationLevelId, "ft");
+            
+            LocationLevelValue valueInMeters = dao.getLatestLocationLevelValue(
+                tx, testLocationLevelId, "m");
+            
+            if (valueInFeet != null && valueInMeters != null)
+            {
+                // Verify units are set correctly
+                assertEquals("ft", valueInFeet.getUnits());
+                assertEquals("m", valueInMeters.getUnits());
+                
+                // Verify conversion is roughly correct (1 ft ≈ 0.3048 m)
+                // Only check if the original values are different from zero
+                if (Math.abs(valueInFeet.getLevelValue()) > 0.001)
+                {
+                    double expectedMeters = valueInFeet.getLevelValue() * 0.3048;
+                    assertEquals(expectedMeters, valueInMeters.getLevelValue(), 0.01,
+                        "Conversion from feet to meters should be accurate");
+                }
+            }
+        }
+        finally
+        {
+            dao.close();
+        }
+    }
+    
+    @Test
+    @EnableIfTsDb({"CWMS-Oracle"})
+    public void testGetLocationLevelSpecs() throws Exception
+    {
+        if (dao == null) 
+        {
+            return; // Skip for non-CWMS databases
+        }
+        
+        String testLocationId = "TEST_LOCATION";
+        
+        try (DataTransaction tx = db.newTransaction())
+        {
+            List<LocationLevelSpec> specs = dao.getLocationLevelSpecs(tx, testLocationId);
+            
+            assertNotNull(specs, "Specs list should not be null");
+            
+            // If specs exist, verify their structure
+            for (LocationLevelSpec spec : specs)
+            {
+                assertNotNull(spec.getLocationLevelId(), "Location level ID should not be null");
+                assertNotNull(spec.getLocationId(), "Location ID should not be null");
+                assertNotNull(spec.getParameterId(), "Parameter ID should not be null");
+            }
+        }
+        finally
+        {
+            dao.close();
+        }
+    }
+    
+    @Test
+    @EnableIfTsDb({"CWMS-Oracle"})
+    public void testTransactionSupport() throws Exception
+    {
+        if (dao == null) 
+        {
+            return; // Skip for non-CWMS databases
+        }
+        
+        String testLocationLevelId = "TEST_LOCATION.Stage.Const.0.Test";
+        String testLocationId = "TEST_LOCATION";
+        
+        DataTransaction tx = null;
+        try
+        {
+            tx = db.newTransaction();
```

Comment:
DataTransaction implements `AutoCloseable` which means you can put it in try-with-resources but it's not going to get flagged if it's not in the try-with.

The root opening of a transaction *SHOULD* be in the try-with-resources.

Reply:
See you're next test in this file.